src/Gui/Placement.cpp

```diff
@@ -559,6 +572,83 @@ void Placement::showDefaultButtons(bool ok)
     }
 }
 
+bool Placement::eventFilter(QObject*, QEvent* ev) {
+    //handle spacenav daemon motion events in placement window
+    if (//event->type() == Spaceball::ButtonEvent::ButtonEventType ||
+        ev->type() == Spaceball::MotionEvent::MotionEventType){
+                Spaceball::MotionEvent *motionEvent = dynamic_cast<Spaceball::MotionEvent*>(ev);
+                if (!motionEvent) {
+                    Base::Console().Log("invalid spaceball motion event in bool Placement::eventFilter(QObject*, QEvent* ev)\n");
+                    return false;
+                    }
+                motionEvent->setHandled(true);
+
+                static double translationConstant(.001f);
+                double xTrans = static_cast<double>(motionEvent->translationX())*translationConstant;
+                double yTrans = static_cast<double>(motionEvent->translationY())*translationConstant;
+                double zTrans = static_cast<double>(motionEvent->translationZ())*translationConstant;
+
+                static double rotationConstant(.001f);
+                double dY=static_cast<double>(motionEvent->rotationZ()) * rotationConstant;
+                double dP=static_cast<double>(motionEvent->rotationY()) * rotationConstant;
+                double dR=static_cast<double>(motionEvent->rotationX()) * rotationConstant;
+
+                if(xTrans == 0 && yTrans == 0 && zTrans == 0 && dY ==0 && dP ==0 && dR ==0 ) return false;
+                signalMapper->blockSignals(true);
+//                printf("\nSpaceball Placement Event\t(x,y,z)=(%f,%f,%f)\t(Rx,Ry,Rz)=(%f,%f,%f);\n",xTrans,yTrans,zTrans,dR,dP,dY  );

```

Comment:
Removed commented code

Reply:
I am developing a 6dof controller and this snippet is useful for debugging purposes.
I am not very familiar how debugging code is usually included in Freecad.
Could you recommend an acceptable way to include it?

Reply:
Disclaimer, I don't know what's preferred in freecad. This is just my personal opinion.

The issue I have with including debug code in a non-personal project, is that the debug code is essentially dead code for everyone except the author. Debug code usually make code more difficult to read and makes it more difficult to maintain and refactor.

When coding, I try to prioritize others time above my own. I'm only one person, but there might be hundreds of people who my laziness affects.

Personally, I do the following:
1. don't stage debug code when creating commits
2. stash and test without debug code before push
3. If I want to keep the debug code, I create a new branch and a a commit the debug code and call it `DEBUG, DONT PUSH` to avoid accidentally create a pull request with that code.

Later when/if I want to continue with the debugging, I can simply checkout and rebase the debug code on latest main.  There might be conflicts this way, but that just shows that it was a good idea to avoid commit that code in the first place.